### ISSUE TITLE
Comments for "Github Issues As Comments"

### ISSUE THREAD

**Issue by kevquirk** [kevquirk/kevq.uk-post-comments#51]

**Comment by Fastidious** [comment]
Wouldn’t this be considered an abuse by GitHub? Just wondering.

**Comment by dasm** [comment]
Ha. Interesting :) 👍

**Comment by benharri** [comment]
have you seen https://utteranc.es/ - similar project with an embeddable widget

**Comment by darekkay** [comment]
There's also [Gitment](https://github.com/imsun/gitment) and [StaticMan](https://staticman.net/), but I guess a DIY solution has its own charm 😉

**Comment by raretrack** [comment]
Well it certainly adds some friction that should put off the bots and spammers :)

**Comment by dasm** [comment]
@raretrack that's true. I would personally prefer more seamless solution, where github issues could be present on website. Did you consider it @kevquirk ?

**Comment by Mr-Steal-Your-Script** [comment]
This seems cool. Maybe I'll add it to my GH page later.

**Comment by kevquirk** [comment]
> Wouldn’t this be considered an abuse by GitHub? Just wondering.

I suppose we will see! 😁

**Comment by kevquirk** [comment]
> There's also [Gitment](https://github.com/imsun/gitment) and [StaticMan](https://staticman.net/), but I guess a DIY solution has its own charm 😉 
> 
> 
> 
> You can have a look at [my blog post](https://darekkay.com/blog/static-site-comments/), where I'm collecting all the blog comment solutions I can find. Ironically, I have disabled the comments on my own blog a few months ago 😅 

Ohhhh I need to look into this. Thanks for linking to this.

**Comment by kevquirk** [comment]
> @raretrack that's true. I would personally prefer more seamless solution, where github issues could be present on website. Did you consider it @kevquirk ?

I did consider it, but I much prefer having comments on a separate platform than cluttering up the blog post.

**Comment by Xerbo** [comment]
Well it's certainly an interesting idea using GitHub issues for comments, but not gonna lie, I dig it

**Comment by kevquirk** [comment]
>  Well it's certainly an interesting idea using GitHub issues for comments, but not gonna lie, I dig it

Seems to be working well. A bonus that I didn't account for is that I can reply to comments using the GitHub app on my phone.

**Comment by AlanSRalph** [comment]
I don't often get comments on my blog posts, but those that I do receive then become part of the website if I choose to let them be so. Yes, WordPress's commenting system has some issues, but none that are dealbreakers for me. And honestly, I'm of an age now where I prefer an easy tech life, even if the tech may sometimes decide otherwise.

I can sort of see the appeal of using GitHub for comments, as a kind of backchannel. My big concern would be that there's now a lot more brittleness in your blogging setup. Future updates to WordPress could break your code linking your blog posts to GitHub. And, while it might not be likely, there is the possibility that GitHub might go away, or decide that they don't want you using their system in the way you're doing now.

**Comment by Xerbo** [comment]
> there is the possibility that GitHub might go away, or decide that they don't want you using their system in the way you're doing now.

Well, that doesn't really matter, it can be done with any git service that supports commenting on issues (GitLab, Gitea, Gogs, etc). GitHub was probably chosen since most people (that read this type of content) already have a GitHub account.

**Comment by kevquirk** [comment]
> My big concern would be that there's now a lot more brittleness in your blogging setup. Future updates to WordPress could break your code linking your blog posts to GitHub. 

It's literally a hyperlink. Yes it's injected with a little bit of php, but there's nothing clever going on. If I were pulling the comments in via and API that could change and break everything, then yeah, I'd be inclined to agree. But we're literally talking about a link here.

> And, while it might not be likely, there is the possibility that GitHub might go away, or decide that they don't want you using their system in the way you're doing now.

I'd be surprised if that happened. If it did though, no big deal. I'd just go without comments. They're more a headache that anything else, so anything I can do to make it easier for me to manage is a win.

This setup, I feel, is a good balance between giving people the option to comment, and me not having all the headaches of managing a commenting system - spam etc.

**Comment by MattMattV** [comment]
Nice idea ! You could have also used comments of a Reddit post on a /r/KevQuirk subreddit by example and you would had hierarchical conversations (:

**Comment by kevquirk** [comment]
> Nice idea ! You could have also used comments of a Reddit post on a /r/KevQuirk subreddit by example and you would had hierarchical conversations (:

Oooo that is a good idea. Although Reddit doesn't have as good of an experience as here. I find the performance to often be slow at best.

**Comment by jmtd** [comment]
This is a terrible idea for more reasons than I can enumerate. But here’s one: I’ve gone to the trouble of logging into GitHub (a site I use frequently on my full size computers) on my mobile, navigating fetching my password that the phone doesn’t know and then 2FA, just to post this comment. I wouldn’t bother to do this for other blog posts. And if I didn’t already have a GitHub account I certainly wouldn’t sign up for on just to comment on a blog.  

What does the issue state (open, closed) mean in the context of a comments thread? What, besides the parent blog post, relates this comment on the issue to any other? It would make marginally more sense if your blog content was in the GitHub repo. But it isn’t!

Really, you’ve implemented this for your ease and not for your readers. Generally you should put readers first.

**Comment by kevquirk** [comment]
> I wouldn’t bother to do this for other blog posts. And if I didn’t already have a GitHub account I certainly wouldn’t sign up for on just to comment on a blog.

So why do it for this post then? You're free not to comment - for people that either don't have a GitHub account, or don't want to use this **option**, I have multiple ways to contact me on my contact for **AND** the guestbook on my site.

> What does the issue state (open, closed) mean in the context of a comments thread? What, besides the parent blog post, relates this comment on the issue to any other? It would make marginally more sense if your blog content was in the GitHub repo. But it isn’t!

Open/Closed of the issue doesn't mean anything in this context.

> Really, you’ve implemented this for your ease and not for your readers. Generally you should put readers first.

I've implemented this for my ease of use **AND** my readers. It's a middle ground. A compromise. Like I said earlier - this is just another option people can use. If they don't want to that's fine. So feel free to shut down your full size computer, log out of your password manager, and put your 2FA token away. It's fine; you don't have to use this option.

**Side note:** Just because *you* don't find this option to be easier, doesn't mean I'm not putting my readers first. It simply means I'm not putting *you* first.

**Comment by Mr-Steal-Your-Script** [comment]
@kevquirk 
Some people like to take the joy out of what others take pride in. I have my own GitHub page, and I wanted to add a comment section to some of my entries on the site. I found [this option](https://cheapskatesguide.org/articles/free-forum.html). It seems nice and cool, but since it uses PHP, I can't use it on GitHub pages. I thought about other ways to do so, and GitHub issues came to mind. I personally would like to thank you for this option, and I think it would be cool if the comments were embedded in the page too.

**Comment by jmtd** [comment]
> So why do it for this post then? You're free not to comment - for people that either don't have a GitHub account, or don't want to use this option, I have multiple ways to contact me on my contact for AND the guestbook on my site.

Because I’ve been a reader of your blog, and occasional commenter, I’d got something out of reading your blog, and I felt like putting in the effort required to point out the downsides of this approach, which I hadn’t seen discussed yet. You invited feedback by blogging specifically about the change. I’m sorry if you don’t like it. It was meant to be constructive, not heated, but your response feels emotional so I’m sorry if you took it a different way. 

> Some people like to take the joy out of what others take pride in

That’s not me at all. Kev takes pride in his blog and I point out what I think is a mistake as a kindness. If I wanted to be snarky I’d be subtweeting about it or something, but I don’t. 

> Just because you don't find this option to be easier, doesn't mean I'm not putting my readers first. It simply means I'm not putting you first.

I don’t think that’s true. Consider all the reasons you ditched Wordpress comments and went down this route: to reduce the likelihood of blog spam, to avoid having to maintain more Wordpress plugins, etc. They’re all effort/pain for you, not your readers. Consider also that the inline comments are right there next to the article, that the user doesn’t need a separate account set up to comment. None of this is specifically about my preferences.

**Comment by kevquirk** [comment]
@jmtd apologies if my response came off as emotional, that wasn't my intention. Constructive criticism is always welcome with me, I just didn't feel yours was all that constructive. Statements like...

> This is a terrible idea for more reasons than I can enumerate.

Are emotional to me, and not very constructive or helpful.

Anyway, I digress...

> Consider all the reasons you ditched Wordpress comments and went down this route...

While that is true, I feel there should be a balance. I spend hours a week maintaining my blog and writing content (both on my blog and my newsletter). These hours spent every week are so that my readers enjoy the experience on my blog.

Sure, not everyone is going to like this method. That's why I offer a Guestbook that has the option to be completely anonymous, as well as an email address (and public key), Mastodon account. Keybase account and even Twitter. Between all of the above I think there's something for everyone there.

There's nothing to stop people coming here and reading the comments, then contacting me using any of the methods listed above. The difference between reading the comments on here and on my blog is literally 1 click and a fraction of as second.
